src/openrct2/scenario/scenario.c

```diff
@@ -660,38 +660,78 @@ uint32 scenario_rand_max(uint32 max)
  * Prepare rides, for the finish five rollercoasters objective.
  *  rct2: 0x006788F7
  */
-static void scenario_prepare_rides_for_save()
+static bool scenario_prepare_rides_for_save()
 {
     sint32 isFiveCoasterObjective = gScenarioObjectiveType == OBJECTIVE_FINISH_5_ROLLERCOASTERS;
+    sint32 i;
+    rct_ride *ride;
+    uint8 rcs = 0;
 
-    // Set all existing track to be indestructible
+    FOR_ALL_RIDES(i, ride)
+    {
+        const rct_ride_entry * rideEntry = get_ride_entry(ride->subtype);
+        
+        // If there are more than 5 roller coasters, only mark the first five.
+        if (isFiveCoasterObjective &&
+            rideEntry != NULL &&
+            ((rideEntry->category[0] == RIDE_GROUP_ROLLERCOASTER || rideEntry->category[1] == RIDE_GROUP_ROLLERCOASTER) &&
+             rcs < 5))
+        {
+            ride->lifecycle_flags |= RIDE_LIFECYCLE_INDESTRUCTIBLE_TRACK;
+            rcs++;
+        }
+        else
+        {
+            ride->lifecycle_flags &= ~RIDE_LIFECYCLE_INDESTRUCTIBLE_TRACK;
+        }
+    }
+
+    if (isFiveCoasterObjective && rcs < 5)
+    {
+        gGameCommandErrorText = STR_NOT_ENOUGH_ROLLER_COASTERS;
+        return false;
+    }
+
+    bool markTrackAsIndestructible;
     map_element_iterator it;
     map_element_iterator_begin(&it);
-    do {
-        if (map_element_get_type(it.element) == MAP_ELEMENT_TYPE_TRACK) {
+    do
+    {
+        if (map_element_get_type(it.element) == MAP_ELEMENT_TYPE_TRACK)
+        {
+            markTrackAsIndestructible = false;
+
             if (isFiveCoasterObjective)
-                it.element->flags |= 0x40;
+            {
+                const rct_ride * ride = get_ride(it.element->properties.track.ride_index);
```

Comment:
You've already declared a variable `ride` above - you'll have to either forgo the declaration here or change the variable name.